src/proto/proto_common.cpp

```diff
@@ -2038,7 +2038,7 @@ void init_data_readers(bool master, const lbann_data::LbannPB& p, std::map<execu
         (*(ascii_reader *)reader_validation) = (*(ascii_reader *)reader);
       }
 
-      reader_validation->swap_role("validate");
+      reader_validation->set_role("validate");
```

Comment:
I am not sure that what you are proposing here is correct.  The validation reader is a copy of the primary reader, but we then need to adjust the sample indices after the copy.

Reply:
By the way,  the adjustment of the sample indices is done by the following line reader_validation->use_unused_index_set() in a non-reversible way. What swap_role does is simply to swap m_use_percent and m_validation_percent, which is not a valid thing to do as these percents are not compatible.